src/lib/auth.ts

```diff
@@ -189,8 +197,21 @@ export interface PortalFirebaseTeacherJWT extends BasePortalFirebaseJWT {
 
 export type PortalFirebaseJWT = PortalFirebaseStudentJWT | PortalFirebaseTeacherJWT;
 
-export const getPortalJWTWithBearerToken = (basePortalUrl: string, type: string, rawToken: string) => {
+export const getPortalJWTWithBearerToken =
+              (basePortalUrl: string, type: string, rawToken?: string): Promise<[string, PortalJWT]> => {
   return new Promise<[string, PortalJWT]>((resolve, reject) => {
+
+    const sessionRawPortalJWT = window.sessionStorage.getItem(rawPortalJWTSessionKey);
```

Comment:
I think there needs to be a little more work here (and for the Firebase JWT storage):

1. We need to check to see if the token is expired, and if it is fall through to requesting a new token.
2. Instead of failing with "Invalid portal token" if the token doesn't decode I think we should fall through to requesting a new token.